scripts/west_commands/run_common.py

```diff
@@ -128,16 +127,19 @@ def _build_dir(args, die_if_none=True):
     if args.build_dir:
         return args.build_dir
 
-    cwd = getcwd()
-    default = path.join(cwd, DEFAULT_BUILD_DIR)
-    if is_zephyr_build(default):
-        return default
-    elif is_zephyr_build(cwd):
-        return cwd
+    dir = find_build_dir(None)
+
+    if dir and is_zephyr_build(dir):
+        return dir
     elif die_if_none:
-        log.die('--build-dir was not given, and neither {} '
-                'nor {} are zephyr build directories.'.
-                format(default, cwd))
+        msg ='--build-dir was not given, '
+        if dir:
+            msg = msg + 'and neither {} nor {} are zephyr build directories.'
+        else:
+            msg = msg + ('{} is not a build directory and the default build '
+                         'directory cannot be determined. Check your '
+                         'build.dir-fmt configuration option')
```

Comment:
no need to make a change, but it would be slightly nicer to print the value here if it is not `None`

Reply:
if what is not `None`? `dir`? I check against that and print it above?

Reply:
`build.dir-fmt`

Reply:
oh right. That would force me to load it from outside `build_helpers.py`. I'd rather not do that.

Reply:
No worries, then.